packages/filesystem/src/node/node-filesystem.ts

```diff
@@ -328,10 +327,9 @@ export class FileSystemNode implements FileSystem {
         };
     }
 
-    protected doCreateDirectoryStat(uri: URI, path: string, stat: fs.Stats, depth: number): FileStat {
-        const files = fs.readdirSync(path);
-        const hasChildren = files.length > 0;
-        const children = hasChildren ? depth > 0 ? this.doGetChildren(uri, files, depth) : undefined : [];
+    protected doCreateDirectoryStat(uri: URI, stat: fs.Stats, depth: number): FileStat {
+        const children = depth > 0 ? this.doGetChildren(uri, depth) : [];
+        const hasChildren = children.length > 0;
```

Comment:
The `hasChildren` value is now always `false`, if `depth === 0`.

Reply:
Yes by design what is the issue ?

Reply:
I see this is used in the move for a special case handling but I think we could do this another way.
And actually remove hasChildren completely

Reply:
Yes, seems like we don't use that property at all.